crates/bevy_core/src/bytes.rs

```diff
@@ -50,9 +50,8 @@ where
 {
     fn from_bytes(bytes: &[u8]) -> Self {
         unsafe {
-            let byte_ptr = bytes.as_ptr();
-            let ptr = byte_ptr as *const Self;
-            (*ptr).clone()
+            let (_, body, _) = bytes.align_to::<Self>();
```

Comment:
`align_to` ignoring the unaligned prefix and postfix seems suspicious here.  Maybe it could use https://doc.rust-lang.org/std/ptr/fn.read_unaligned.html instead?

Reply:
Yep, that makes more sense.
I update the PR :)